tensorflow/python/ops/random_ops.py

```diff
@@ -300,6 +300,8 @@ def random_uniform(shape,
         if not maxval_is_one:
           result = math_ops.multiply(result, maxval)
       else:
+        maxval = array_ops.ensure_shape(maxval, ())
```

Comment:
Doesn't this prevent us from making broadcast-correct randomuniform? As in, if I'm asking for a random uniform with shape [a, b] I should be able to have maxval/minval with any shape which broadcasts to [a, b] as we can broadcast the elementwise operations